src/main/java/teammates/storage/sqlapi/CoursesDb.java

```diff
@@ -146,6 +146,19 @@ public Section getSectionByCourseIdAndTeam(String courseId, String teamName) {
         return HibernateUtil.createQuery(cr).getResultStream().findFirst().orElse(null);
     }
 
+    /**
+     * Deletes all sections by {@code courseId}.
+     */
+    public void deleteSectionsByCourseId(String courseId) {
+        CriteriaBuilder cb = HibernateUtil.getCriteriaBuilder();
+        CriteriaQuery<Section> cq = cb.createQuery(Section.class);
+        Root<Section> sRoot = cq.from(Section.class);
+        Join<Section, Course> sJoin = sRoot.join("course");
+        cq.select(sRoot).where(cb.equal(sJoin.get("id"), courseId));
+        List<Section> sectionsToDelete = HibernateUtil.createQuery(cq).getResultList();
+        sectionsToDelete.forEach(HibernateUtil::remove);
```

Comment:
If we want to do this, I think we can delete in batches (unless there's a reason why we're not doing that). I think it would be something like this:

```suggestion
        CriteriaDelete<Section> cd = cb.createCriteriaDelete(Section.class);
        Root<Section> sRoot = cd.from(Section.class);
        Join<Section, Course> sJoin = sRoot.join("course");
        cd.where(cb.equal(sJoin.get("id"), courseId));
        HibernateUtil.createMutationQuery(cd).executeUpdate();
```

We may need to add a method in `HibernateUtil` to call `Session::createMutationQuery`.

See also:
- [java - How to delete by criteria in hibernate? - StackOverflow](https://stackoverflow.com/a/54548571)
- [Criteria Update/Delete – The easy way to implement bulk operations with JPA2.1](https://thorben-janssen.com/criteria-updatedelete-easy-way-to/)

Reply:
Wow Jay, this is very cool! Thanks Jay! Have updated the code and added a static method in `HibernateUtil`.

Reply:
@TEAMMATES/cs3281-2023-students Hey guys, adding yall for visibility on this batch update/delete that Hibernate provides.